src/h3/_cy/cells.pyx

```diff
@@ -305,53 +301,51 @@ cpdef double mean_hex_area(int resolution, unit='km^2') except -1:
     try:
         area *= convert[unit]
     except:
-        raise H3ValueError('Unknown unit: {}'.format(unit))
+        raise ValueError('Unknown unit: {}'.format(unit))
 
     return area
 
 
 cpdef double cell_area(H3int h, unit='km^2') except -1:
     cdef:
-        h3lib.H3Error err
         double area
 
-    check_cell(h)
-
     if unit == 'rads^2':
         err = h3lib.cellAreaRads2(h, &area)
     elif unit == 'km^2':
         err = h3lib.cellAreaKm2(h, &area)
     elif unit == 'm^2':
         err = h3lib.cellAreaM2(h, &area)
     else:
-        raise H3ValueError('Unknown unit: {}'.format(unit))
+        raise ValueError('Unknown unit: {}'.format(unit))
+
+    check_for_error(err)
 
     return area
 
 
+cdef couldnt_find_line(err, start, end):
+    msg = "Couldn't find line between cells {} and {}"
+    msg = msg.format(hex(start), hex(end))
+
+    check_for_error_msg(err, msg)
+
 cpdef H3int[:] line(H3int start, H3int end):
     cdef:
-        h3lib.H3Error err
         int64_t n
 
-    check_cell(start)
-    check_cell(end)
-
+    # todo: can we segfault here with invalid inputs?
+    # Can we trust the c library to validate the start/end cells?
+    # probably applies to all size/work pairs of functions...
     err = h3lib.gridPathCellsSize(start, end, &n)
```

Comment:
I'm not sure what this comments means

Reply:
Is it possible to segfault with a pair of calls to `gridPathCellsSize`/`gridPathCells`? If this function were fuzzed, for example?

If we can't segfault and just get junk output, then for the Python bindings we have a choice to validate or not validate the inputs for the user.

However, if it is possible to segfault, we have no choice and we *must* validate the inputs well enough to guarantee that a segfault can't happen.